java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1157,62 +1157,61 @@ private void handleImageBuildData(ServerAction serverAction, JsonElement jsonRes
-            details.ifPresentOrElse(det -> {
-                Optional<ImageProfile> profileOpt =
-                        ImageProfileFactory.lookupById(det.getImageProfileId());
-
-                profileOpt.ifPresent(p -> p.asKiwiProfile().ifPresent(kiwiProfile -> {
-                    serverAction.getServer().asMinionServer().ifPresent(minionServer -> {
-                        // Download the built Kiwi image to SUSE Manager server
-                        OSImageInspectSlsResult.Bundle bundleInfo =
-                                Json.GSON.fromJson(jsonResult, OSImageBuildSlsResult.class)
-                                        .getKiwiBuildInfo().getChanges().getRet().getBundle();
-                        infoOpt.ifPresent(info -> info.setChecksum(
-                                ImageInfoFactory.convertChecksum(bundleInfo.getChecksum())));
-                        MgrUtilRunner.ExecResult collectResult = systemQuery
-                                .collectKiwiImage(minionServer, bundleInfo.getFilepath(),
-                                        OSImageStoreUtils.getOsImageStorePath() + kiwiProfile.getTargetStore().getUri())
-                                .orElseThrow(() -> new RuntimeException("Failed to download image."));
-
-                        if (collectResult.getReturnCode() != 0) {
-                            serverAction.setStatus(ActionFactory.STATUS_FAILED);
-                            serverAction.setResultMsg(StringUtils
-                                    .left(printStdMessages(collectResult.getStderr(), collectResult.getStdout()),
-                                            1024));
-                        }
-                    });
-                }));
-                ImageInspectAction iAction = ActionManager.scheduleImageInspect(
-                        action.getSchedulerUser(),
-                        action.getServerActions()
-                                .stream()
-                                .map(ServerAction::getServerId)
-                                .collect(Collectors.toList()),
-                        Optional.of(action.getId()),
-                        det.getVersion(),
-                        profileOpt.map(ImageProfile::getLabel).orElse(null),
-                        profileOpt.map(ImageProfile::getTargetStore).orElse(null),
-                        Date.from(Instant.now())
-                );
-                try {
-                    TASKOMATIC_API.scheduleActionExecution(iAction);
-                }
-                catch (TaskomaticApiException e) {
-                    LOG.error("Could not schedule image inspection");
-                    LOG.error(e);
-                }
-
-                infoOpt.ifPresent(info -> {
-                    info.setRevisionNumber(info.getRevisionNumber() + 1);
-                    info.setInspectAction(iAction);
-                    ImageInfoFactory.save(info);
+            if (details.isEmpty()) {
```

Comment:
make sense. My first implementation had Optional in ImageBuildAction class (and probably it would be the best choice), but then due to problem with hibernate, I remove it but leave it here. I don't have a strong opinion for that, so it's fine for me to avoid wrapping